CodeStarterCamp_Week2/main.swift

```diff
@@ -8,5 +8,43 @@
 
 import Foundation
 
-print("Hello, World!")
+//찍은 번호
+let myLottoNumbers: [Int] = [4, 13, 24, 36, 38, 45]
 
+//로또 번호 배열
+var lottoNumbers: [Int] = []
+
+//로또 번호 랜덤 뽑기
+func pickLottoNumbers() {
+    while lottoNumbers.count < 6 {
+        let numbers = Int.random(in: 1...45)
+        if !lottoNumbers.contains(numbers) {
+            lottoNumbers.append(numbers)
+        }
+    }
+}
+
+pickLottoNumbers()
+
+//찍은 번호와 로또 번호가 일치하면 들어가는 배열
+var coincideNumbers: [Int] = []
```

Comment:
처음에 이 네이밍을 보고, 솔직히 이렇게 생각했습니다.
coincide 는 뭐지..? `우연`이란 뜻의 coincidence 단어는 아는데.. 그거의 동사원형인가보네.
그러면 "우연한 숫자들" 인가? ...

네이밍은 정말 어렵죠. 🤔 
저도 항상 어려워하지만, 그래도 개인적으로 이런 생각을 하면서 지으려고 해요.

"자주 안 쓰이거나 난이도가 높은 단어보다는, 사람들에게 익숙한 단어를 사용하면서, 코드의 `역할`을 충분히 담아내자."

주석에 솜이 직접 `"일치하면"` 이라고 쓰셨잖아요?
저는 저 표현을 보고 `match` 라는 단어가 떠올랐어요.

하지만 저 동사원형 그대로 `matchNumbers` 라고 쓰면.. 상수/변수 네이밍이 아니라, 마치 `메서드` 네이밍 같아지죠.
그러니까 동사에 -ing 를 붙이거나 -ed 를 붙여서 `분사` 형태로 만들어서 쓰면 될 것 같아요.

숫자들 입장에선 `매칭을 당한 것` 이기 때문에 피동의 늬앙스를 담아 `matched` 로 표현하면 되겠군요. 😄 

```suggestion
//찍은 번호와 로또 번호가 일치하면 들어가는 배열
var matchedNumbers: [Int] = []
```

Reply:
이 배열 이름 짓는 것에서 저도 좀 고민을 많이 했습니다. match(비교하다) 자체는 이미 함수명으로 쓰고 있어서 배열명으로 쓰자니 제 스스로 헷갈리더라고요. 그래서 이 배열에는 앞의 두 배열의 숫자가 *일치하면* 추가가 되게 설정을 해둬서 일치하다, 라는 coincide 라는 명을 붙였습니다. 저도 구글에서 찾아본거고 처음 보는 영단어라ㅎㅎ 다른 분들이 혼란이 오실 거 같아요! 
근데 예거처럼 ed를 붙인다면 덜 헷갈릴까요?? 변수명을 제가 직접 치는 것보다 Xcode 내의 자동완성 기능을 사용할 때가 많아서 조금 고민이네요🧐
수정했습니다:)

Reply:
`matchedNumbers` 변수와 `matchNumber` 라는 메서드의 이름이 유사하기 때문에, 추후 더 헷갈리지 않을까 고민하셨군요!
반드시 필요한 고민이구요. 오히려 마냥 수용하지 않고 이렇게 한 번 더 고민해보시는 점 너무나 좋습니다. 😄 

저는 `하나의 기능`을 이루는 요소들이라면 비슷한 이름이나 알파벳으로 시작하는 게 '오히려' 좋다고 생각해요. ㅎㅎㅎ
`자동완성` 이야기 해주셨는데, 자동완성 때문에 더더욱 그래요!

이렇게 생각해볼게요...
Xcode 의 `main.swift` 파일에서 `"mat"` 을 쳤어요. (여기까지 쳐야 자동완성이 뜨네요!)
그러면, 아래 캡쳐와 같이 변수와 메서드가 동시에 보여지죠? 

<img width="448" alt="image" src="https://user-images.githubusercontent.com/71127966/164654975-9d8f931b-19f4-4fcc-9176-42983422c966.png">

지금이야 둘의 네이밍이 비슷해서 혼란스러우실 수 있지만...
코드를 더 많이 작성하다 보면, 전~~혀 헷갈리지 않고 오히려 같은 기능을 이루는 2가지 요소가 함께 자동완성되게 만든 개발자의 센스를 고마워하게(...) 되실 수도 있습니다. 😅 

좀 더 팁을 드리자면요!
이렇게 파악하는 것도 도움이 돼요.

- 네이밍 끝에 괄호`(`가 있네? → `메서드`
  - 심지어 네이밍 왼쪽에 초록색 박스 🟩 안에 function 을 의미하는 **_f_** 글자가 찍혀있죠? ㅎㅎ

- 그럼 초록색 박스 안에 있는 `L`은 뭘까요? → "Local variable" 을 의미한다고 합니다. [출처](https://stackoverflow.com/questions/6662395/xcode-intellisense-meaning-of-letters-in-colored-boxes-like-f-t-c-m-p-c-k-etc%29)
  - 참고로, 저장(stored) 프로퍼티와 연산(computed) 프로퍼티 모두 초록색 `L`로 찍힙니다.
  - 그럼 어떻게 구분할까요? 네이밍을 명확하게 해서 딱 봐도 구분할 수 있게 하거나, `Documentation Comments` 기능을 이용하는 겁니다.

- [Documentation Comments](https://sarunw.com/posts/swift-documentation/) → 링크는 다 읽지 말고 간단하게만 슥 보세요.
  - 이거는 사용 방법과 그 결과가 매우 직관적이니까, 아래 코드 예시랑 캡처만 봐도 충분합니다.
  - Xcode 에서 주석은 `//` 죠? 여기에 슬래시 하나 추가해서 3개로 `///` 만들어보세요 😄 
  - 그리고 어딘가에서 그 변수를 자동완성시켜보세요. ㅎㅎ

```swift
/// 찍은 번호와 로또 번호가 일치하면 들어가는 배열
var matchedNumbers: [Int] = []
```

![image](https://user-images.githubusercontent.com/71127966/164674622-4320f8f5-61fe-4950-beb6-e4dd05b42c5e.png)